File: modules/processing/flink-submission-service/app/utils/logger.py
"""
app/utils/logger.py – Shared logger helpers for the Flink submission worker.
"""

from __future__ import annotations

import logging

from . import config


def configure_logging() -> None:
    """Configure the root logger for the submission worker."""
    logging.basicConfig(
        level=logging.DEBUG if config.LOG_LEVEL == "DEV" else logging.INFO,
        format="%(asctime)s  %(levelname)-8s  %(name)s  %(message)s",
    )


def get_logger(name: str) -> logging.Logger:
    """Return a module logger after ensuring logging is configured."""
    configure_logging()
    return logging.getLogger(name)

File: packages/storage/timescale/src/logger.ts
type LogLevel = "debug" | "info" | "warn" | "error";

function log(level: LogLevel, message: string, meta?: unknown) {
    const method = level === "error" ? console.error : level === "warn" ? console.warn : console.log;

    if (meta === undefined) {
        method(`[${level}] ${message}`);
        return;
    }

    method(`[${level}] ${message}`, meta);
}

export const logger = {
    debug(message: string, meta?: unknown) {
        log("debug", message, meta);
    },
    info(message: string, meta?: unknown) {
        log("info", message, meta);
    },
    warn(message: string, meta?: unknown) {
        log("warn", message, meta);
    },
    error(message: string, meta?: unknown) {
        log("error", message, meta);
    }
};